bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -155,6 +155,39 @@ public ImageData[] load(InputStream stream) {
 	return data;
 }
 
+/**
+ * Loads an array of <code>ImageData</code> objects from the
+ * specified input stream. If the stream is a SVG File and zoom is not 0,
+ * this method will try to rasterize the SVG.
+ * Throws an error if either an error occurs while loading the images, or if the images are not
+ * of a supported type. Returns the loaded image data array.
+ *
+ * @param stream the input stream to load the images from
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * @return an array of <code>ImageData</code> objects loaded from the specified input stream
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the stream is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the stream</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image stream contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image stream contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData[] load(InputStream stream, int zoom) {
+	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+    reset();
+    data = FileFormat.load(stream, zoom, this);
+    if (data == null) {
+    	load(stream);
+    }
```

Comment:
This should be moved into the new `FileFormat.load(InputStream, int, ImageLoader)` method (that's why this is passed :) ) and it should be called if within `FileFormat.load()` if it is detected that the passed stream doesn't have SVG content.
This way we can handle the marking and potential going back in the stream in that method (either by wrapping the stream into a `BufferedStream` or using a `PushbackInputStream`.

```suggestion
```